src/mono/mono/mini/mini-wasm-debugger.c

```diff
@@ -1611,6 +1802,65 @@ mono_wasm_invoke_getter_on_object (int object_id, const char* name)
 	return invoke_getter (obj, mono_object_class (obj), name);
 }
 
+EMSCRIPTEN_KEEPALIVE gboolean
+mono_wasm_set_value_on_object (int object_id, const char* name, const char* value)
+{
+	PRINT_DEBUG_MSG (1,  "mono_wasm_set_value_on_object %d, name: %s, value: %s\n", object_id, name, value);
+	MonoObject *obj = get_object_from_id (object_id);
+	
+	if (!obj || !name) {
+		PRINT_DEBUG_MSG (2, "mono_wasm_set_value_on_object: none of the arguments can be null");
+		return FALSE;
+	}
+	MonoClass* klass = mono_object_class (obj);
+
+	gpointer iter;
+handle_parent:
+	iter = NULL;
+	MonoClassField *f;
+	while ((f = mono_class_get_fields_internal (klass, &iter))) {
+		if (!f->name || strcasecmp (f->name, name) != 0)
+			continue;
+		guint8 *val_buf = (guint8 *)g_alloca (mono_class_instance_size (mono_class_from_mono_type_internal (f->type)));
+	
+		if (!decode_value(f->type, val_buf, value)) {
+			return FALSE;
+		}		
+		mono_de_set_interp_var (f->type, (guint8*)obj + f->offset, val_buf);
```

Comment:
`val_buf` can be freed after this, right? The value gets memcpy'd to the target.
The same should be done after `mono_de_set_interp_var(..., val_buf)` in other places.